src/shims/foreign_items/posix/linux.rs

```diff
@@ -0,0 +1,99 @@
+use crate::*;
+use rustc::mir;
+
+impl<'mir, 'tcx> EvalContextExt<'mir, 'tcx> for crate::MiriEvalContext<'mir, 'tcx> {}
+pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx> {
+    fn emulate_foreign_item_by_name(
+        &mut self,
+        link_name: &str,
+        args: &[OpTy<'tcx, Tag>],
+        dest: PlaceTy<'tcx, Tag>,
+        _ret: mir::BasicBlock,
+    ) -> InterpResult<'tcx, bool> {
+        let this = self.eval_context_mut();
+
+        match link_name {
+            "__errno_location" => {
+                let errno_place = this.machine.last_error.unwrap();
+                this.write_scalar(errno_place.to_ref().to_scalar()?, dest)?;
+            }
+
+            // File related shims
+            "close" => {
```

Comment:
Why is close the only linux-specific one? That seems odd.^^

Reply:
it's just because for linux the name is `close` and for macos is `close$NOCANCEL`

Reply:
Well but the same is true for `stat`, `lstat`, `fstat`, right?

Ah but for those we have `statx` as a syscall below, I see.
Please add a comment along the lines of what you wrote here.